### PR TITLE
feat: rework on existing mention-each command

### PR DESCRIPTION
**Pull request by vvaibhavdesai**
- Fixes the structure of the existing command 
- rework on test 

new structure of the command 
image

Ticket: https://github.com/Real-Dev-Squad/discord-slash-commands/issues/75

I get the test coverage is low will increase that in another PR

image

### CODE REVIEW

File: tests/unit/handlers/mentionEachUser.test.ts [hunk 1]
```diff
@@ -9,27 +9,40 @@ describe("Test mention each function", () => {
       DISCORD_GUILD_ID: "123",
       DISCORD_TOKEN: "abc",
     };
-    const optionsArray = [
-      {
-        name: "user",
-        type: 9,
-        value: "860900892193456149",
+
+    const transformedArgument = {
```
Comment (DashDeipayan) [inline tests/unit/handlers/mentionEachUser.test.ts:13]: I think this should go in a fixture

**Review by DashDeipayan — CHANGES_REQUESTED** [review]

File: src/controllers/baseHandler.ts [hunk 2]
```diff
@@ -36,14 +36,12 @@ export async function baseHandler(
     }
     case getCommandName(MENTION_EACH): {
       const data = message.data?.options as Array<messageRequestDataOptions>;
+      const transformedArgument = {
+        roleToBeTaggedObj: data[0],
+        displayMessageObj: data[1] ?? {},
```
Comment (RitikJaiswal75) [inline src/controllers/baseHandler.ts:41]: if i am not mistaken data[0]is the roles option in command and data[1] is the message we want to send, Dould we maybe mention it somewhere or rename the variables

File: tests/unit/handlers/mentionEachUser.test.ts (hunk 1 above)
Comment (RitikJaiswal75) [reply tests/unit/handlers/mentionEachUser.test.ts:13]: +1

File: tests/unit/handlers/mentionEachUser.test.ts [hunk 3]
```diff
@@ -9,27 +9,40 @@ describe("Test mention each function", () => {
...
+  });
+
+  it("should run without displayMessageObj argument", () => {
+    const env = {
+      BOT_PUBLIC_KEY: "xyz",
+      DISCORD_GUILD_ID: "123",
+      DISCORD_TOKEN: "abc",
+    };
+
+    const transformedArgument = {
```
[the remark this answers is not in the document]
Comment (RitikJaiswal75) [reply tests/unit/handlers/mentionEachUser.test.ts:33]: +1

File: src/controllers/baseHandler.ts (hunk 2 above)
Comment (vvaibhavdesai) [reply src/controllers/baseHandler.ts:43]: hey ritik added comments for this.

**Review by RitikJaiswal75 — APPROVED** [review]

**Review by DashDeipayan — APPROVED** [review]
LGTM

**Pull request merged**